src/components/SelectCategory.tsx

```diff
@@ -1,12 +1,15 @@
 import React from "react";
 import { CATEGORIES } from "../constants";
+import { useNavigate } from "react-router-dom"; // Import useNavigate
 
 interface SelectCategoryProps {
   selectQuiz: (category: string, index: number) => void;
   startRandomQuiz: () => void;
 }
 
 const SelectCategory: React.FC<SelectCategoryProps> = SelectCategoryProps => {
+  const navigate = useNavigate(); // React Router navigation
```

Comment:
You can't have `useNavigate` inside of this component. It's only inside of `Router` components that this method can be used.